Source/WarpX.cpp

```diff
@@ -489,6 +503,61 @@ WarpX::PrintGlobalWarnings(const std::string& when)
     amrex::Print() << warn_string;
 }
 
+void
+WarpX::SignalSetFlag(int signal_number)
+{
+    signal_received_flags[signal_number] = true;
```

Comment:
https://en.cppreference.com/w/cpp/utility/program/signal
Since C++11, operations on `atomic` variables of lock-free types are also acceptable. I chose to go that direction because it presents far fewer unique subtleties for other developers than `volatile`